File: Universidad de Granada/Programación y Diseño Orientada a Objetos/QytetetRuby/lib/prueba_qytetet.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

#authors: Miguel Ángel Benítez Alguacil
#         Najib Saadouli Arco

require_relative "qytetet"


module ModeloQytetet
  class PruebaQytetet
    @@juego = Qytetet.instance

    #Método1 muestra las sorpresas cuyo valor sea mayor que 0
    def self.metodo1()
      metodo = Array.new                  #Creamos un array que contendrá las cartas sorpresa que buscamos
      @@juego.mazo.each do |surprise|     #Recorre todo el array cogiendo el valor de cada sorpresa del mazo en cada iteración
        if (surprise.valor > 0)           #Compara el valor de la sorpresa para ver si es mayor que 0
          metodo << surprise
        end
      end
      return metodo
    end

    #Metodo2 muestra las sorpresas de tipo IRACASILLA
    def self.metodo2()
      metodo = Array.new
      @@juego.mazo.each do |surprise|
        if surprise.tipo == TipoSorpresa::IRACASILLA    #Compara si el tipo de la sorpresa es IRACASILLA
          metodo << surprise
        end
      end
      return metodo
    end

    #Metodo3 muestra las sorpresas del tipo que se le indica en el argumento
    def self.metodo3(tipo)
      metodo = Array.new
      @@juego.mazo.each do |surprise|
        if surprise.tipo == tipo              #Compara los tipos de las sorpresas y si es igual al del argumento la añade al array
          metodo << surprise
        end
      end
      return metodo
    end

    
    #Metodo para la entrada por teclado
    def self.getNombreJugadores()
      nombre = Array.new                   #@nombre será nuestro array para almacenar los nombres de los jugadores
      numeroJugadores = 0
      
      loop do
        puts "Introduzca el numero de Jugadores: "
        numeroJugadores = gets.chomp.to_i   #Se hacen lecturas hasta que se cumpla la condición
        break if numeroJugadores >= 2 && numeroJugadores <= @@juego.getMAXJUGADORES  
      end
      
      for i in 0..numeroJugadores-1
        puts "\tEl nombre del jugador #{i} es: "
        nombre << gets.chomp.to_s        #Va introduciendo el nombre de cada jugador
      end
      
      nombre                        #Devuelve el array de nombres
    end
 
    #Bucle while para hacer el getNombreJugadores, alternativa a do while    
=begin      
      No se como hacer referencia a MAX_JUGADORES de Qytetet
      while not numeroJugadores >= 2 && numeroJugadores < 4 Qytetet::MAX_JUGADORES no se inicializa, hay que ponerla final
        puts "Introduzca el numero de Jugadores: "
        numeroJugadores = gets.chomp.to_i  Se hacen lecturas hasta que se cumpla la condición
      end
=end
    
    
    #Main
    def self.main()
      @@juego.inicializarJuego(PruebaQytetet.getNombreJugadores())
=begin
      puts "Metodo 1 -> Las sorpresas con valores mayores que 0 son: \n"
      metodo1().each do |metodo|      Bucle que recorre el mazo de sorpresas y realiza el método 1
        puts metodo.to_s
      end
      puts "\n"
      
      puts "Metodo 2 -> Las sorpresas para ir a casilla son: \n"
      metodo2().each do |metodo|      Bucle que recorre el mazo de sorpresas y realiza el método 2
        puts metodo.to_s
      end
      puts "\n"
      
      puts "Metodo 3 -> Cartas sorpresa separadas por tipo: \n"
      TipoSorpresa::constants.each do |tipo|    Con TipoSorpresa::constants.each obtenemos cada uno de los valores posibles de TipoSorpresa en "tipo"
        puts "Metodos de #{TipoSorpresa.const_get(tipo)}: "
        metodo3(TipoSorpresa.const_get(tipo)).each do |metodo|   Bucle que recorre el mazo de sorpresas y realiza el método 3 para cada uno de los valores de sorpresa
          puts metodo.to_s
        end
      end
     
        
      texto = "\nOPCCIONES PARA EL TEST DE LA PRACTICA 3: "
      texto << "\n 1) Prueba metodo mover y comprar \n "
      texto << "\n 2) Prueba metodo comprar y pagarAlquiler \n "
      texto << "\n 3) Prueba metodo aplicarSorpresa \n "
      texto << "\n 4) Prueba hipotecar, cancelar,vender y edificar casas y hoteles\n"
      texto << "\n 5) Prueba a salir de carcel\n"
      texto << "\n 6) Prueba de ranking y simulacion de @@juego"

      puts texto

      puts "\nIntroduzca la opccion para su prueba: \n"

      test = gets.chomp.to_i

      if (test == 1)
        puts "\n\nSalida del jugador \n"
        puts @@juego.obtenerCasillaJugadorActual.tipo
        puts "\nMoviendo jugador a casilla 5 de calle y comprandola \n" 
        @@juego.mover(5)  #Calle, coste: 650
        puts @@juego.obtenerCasillaJugadorActual.tipo
        puts @@juego.estado
        if (@@juego.estado == EstadoJuego::JA_PUEDECOMPRAROGESTIONAR)
          @@juego.comprarTituloPropiedad 
        end
        puts @@juego.estado
        puts "\nJugador ahora en calle impuesto \n"            
        @@juego.mover(2)  #Impuesto -500
        puts @@juego.obtenerCasillaJugadorActual.tipo
        puts @@juego.estado             
        puts "\nJugador ahora en calle sorpresa \n"            
        @@juego.mover(3)
        puts @@juego.obtenerCasillaJugadorActual.tipo
        puts @@juego.estado 
        puts "\nMostrando propiedades de jugador \n"            
        puts @@juego.jugadorActual.propiedades
      end

      if (test == 2)
        puts "\nMoviendo jugador a casilla 5 de calle y comprandola"            
        puts "\n#{@@juego.jugadorActual.nombre}\n" 
        puts @@juego.obtenerCasillaJugadorActual.numeroCasilla
        @@juego.mover(5)
        puts @@juego.obtenerCasillaJugadorActual.tipo
        puts @@juego.estado
          
        if(@@juego.estado == EstadoJuego::JA_PUEDECOMPRAROGESTIONAR)
          puts @@juego.comprarTituloPropiedad() 
        end
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo} \n" 

        puts "\nMoviendo siguiente jugador a casilla 5 de calle y pagando alquiler \n"             

        @@juego.siguienteJugador()
        puts "\n#{@@juego.jugadorActual.nombre}"                        
        puts "\nSaldo jugador antes de caer = #{@@juego.jugadorActual.saldo} \n" 
        puts @@juego.obtenerCasillaJugadorActual.numeroCasilla                       
        @@juego.mover(5)
        puts @@juego.obtenerCasillaJugadorActual.numeroCasilla                       
        puts @@juego.estado
        puts "\nSaldo jugador después = #{@@juego.jugadorActual.saldo}"                        
        @@juego.siguienteJugador()
        puts "\n#{@@juego.jugadorActual.nombre}\n"                        
        puts "\nSaldo jugador = #{@@juego.jugadorActual.saldo} \n"                               
      end

      if (test == 3)
        puts "\nVamos a recorrer todo el mazo, aplicando así todas las sorpresas: "
        for i in 0..@@juego.getNUMSORPRESAS-1
          @@juego.mover(3)
          puts "\nSORPRESA Nº: #{i}  #{@@juego.cartaActual}"
          @@juego.aplicarSorpresa()
          puts "\nEstado del juego: "
          puts @@juego.estado  
          puts "\nJugador: #{@@juego.jugadorActual}"
          @@juego.siguienteJugador()
        end 
      end

      if(test == 4)   
        puts "\nMoviendo jugador a casilla 4 de calle y comprandola\n"             
        @@juego.mover(4)
        puts @@juego.obtenerCasillaJugadorActual.tipo
        puts @@juego.estado 
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"
        puts "\nEdificando casas y edificando hotel\n"  
        @@juego.comprarTituloPropiedad()
        puts "\n El precio de edificar es: #{@@juego.obtenerCasillaJugadorActual.titulo.precioEdificar}"
        @@juego.edificarCasa(4)
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"
        @@juego.edificarCasa(4)
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"
        @@juego.edificarCasa(4)
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"

        puts "numero de casas = #{@@juego.jugadorActual.propiedades.at(0).numCasas}"

        @@juego.edificarCasa(4)
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"
        @@juego.edificarHotel(4)

        puts "\nEdificado un hotel \n"  
        puts "numero de casas = #{@@juego.jugadorActual.propiedades.at(0).numCasas}"
        puts "numero de hoteles = #{@@juego.jugadorActual.propiedades.at(0).numHoteles}"

        puts "Saldo jugador = #{@@juego.jugadorActual.saldo} \n"
        puts "\nHipotecar\n"
        @@juego.hipotecarPropiedad(4)
        puts "hipotecado : #{@@juego.obtenerCasillaJugadorActual.titulo.hipotecada}"
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"  

        puts "\nCancelando hipoteca"
        @@juego.cancelarHipoteca(4)
        puts "hipotecado : #{@@juego.obtenerCasillaJugadorActual.titulo.hipotecada}"
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"

        puts "\nVenta propiedad"
        @@juego.venderPropiedad(4)
        puts "propietario #{@@juego.obtenerCasillaJugadorActual.titulo.tengoPropietario()}"
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"
      end

      if(test == 5)
        puts "\nSe mete al jugador en la carcel\n"
        @@juego.mover(12)
        puts "encarcelado = #{@@juego.jugadorActual.encarcelado}"
        @@juego.intentarSalirCarcel(MetodoSalirCarcel::PAGANDOLIBERTAD)
        puts "encarcelado = #{@@juego.jugadorActual.encarcelado}"
        puts "Saldo jugador = #{@@juego.jugadorActual.saldo}\n"

        fuera = false
        puts "\nSe vuelve a meter al jugador en la carcel\n"
        @@juego.mover(12)
        puts "encarcelado = #{@@juego.jugadorActual.encarcelado}"
        while(!fuera)
          @@juego.intentarSalirCarcel(MetodoSalirCarcel::TIRANDODADO)
          if(!@@juego.jugadorActual.encarcelado)
            fuera = true 
          end
        end
        puts "encarcelado = #{@@juego.jugadorActual.encarcelado}"
        puts "Saldo jugador =  #{@@juego.jugadorActual.saldo} \n"
      end

      if(test == 6)
=begin
        for i in 0...30
          @@juego.jugar()
          @@juego.siguienteJugador()
        end

        @@juego.mover(2)
        @@juego.siguienteJugador()
        @@juego.mover(13)
        @@juego.comprarTituloPropiedad()
        @@juego.edificarCasa(13)
        @@juego.edificarCasa(13)
        @@juego.edificarCasa(13)
        @@juego.hipotecarPropiedad(13)
        @@juego.siguienteJugador()
        @@juego.mover(6)
        @@juego.comprarTituloPropiedad()
        @@juego.siguienteJugador()
        @@juego.mover(4)
        @@juego.comprarTituloPropiedad()
        @@juego.siguienteJugador()
        @@juego.mover(3)
        @@juego.aplicarSorpresa()
        @@juego.mover(2)
        puts "\nEl capital del jugador #{@@juego.jugadorActual.nombre} es: #{@@juego.jugadorActual.obtenerCapital}\n"
        @@juego.siguienteJugador()
        @@juego.mover(2)
        puts "\nEl capital del jugador #{@@juego.jugadorActual.nombre} es: #{@@juego.jugadorActual.obtenerCapital}\n"
        @@juego.obtenerRanking()
        puts "\nEl ganador es el jugador #{@@juego.jugadores.at(0).nombre} con un capital de #{@@juego.jugadores.at(0).obtenerCapital}\n"
        @@juego.jugadores.each do |jugador|
          puts jugador
        end
      end
=end 
      @@juego.mover(3)
      
      puts @@juego.jugadorActual.to_s << "\n"
      
      @@juego.aplicarSorpresa
      
      puts @@juego.jugadorActual.to_s << "\n"
      
      for i in 0..10 do
        if(@@juego.jugadorActual.casillaActual.numeroCasilla == 0)
          @@juego.mover(10)
        end
        if(@@juego.jugadorActual.casillaActual.numeroCasilla != 0)
          @@juego.mover(0)
        end
      end
      
      @@juego.mover(4)
      
      if(@@juego.comprarTituloPropiedad)
        puts @@juego.jugadorActual.to_s << "\n"
      end
      
      for i in 0..10
        if(@@juego.edificarCasa(4))
          puts @@juego.jugadorActual.to_s << "\n"
        end
        for j in 0..10
          if(@@juego.edificarHotel(4))
            puts @@juego.jugadorActual.to_s << "\n"
          end
          for k in 0..10
            if(@@juego.edificarCasa(4))
              puts @@juego.jugadorActual.to_s << "\n"
            end
          end
        end
      end
      
      @@juego.siguienteJugador
      
      puts @@juego.jugadorActual.to_s << "\n"
      
      for i in 0..10 do
        if(@@juego.jugadorActual.casillaActual.numeroCasilla == 0)
          @@juego.mover(10)
        end
        if(@@juego.jugadorActual.casillaActual.numeroCasilla != 0)
          @@juego.mover(0)
        end
      end
      
      @@juego.mover(5)
      
      if(@@juego.comprarTituloPropiedad)
        puts @@juego.jugadorActual.to_s << "\n"
      end
      
      for i in 0..10
        if(@@juego.edificarCasa(5))
          puts @@juego.jugadorActual.to_s << "\n"
        end
        for j in 0..10
          if(@@juego.edificarHotel(5))
            puts @@juego.jugadorActual.to_s << "\n"
          end
          for k in 0..10
            if(@@juego.edificarCasa(5))
              puts @@juego.jugadorActual.to_s << "\n"
            end
          end
        end
      end
      
      @@juego.siguienteJugador
      puts @@juego.jugadorActual.to_s << "\n"
      @@juego.mover(2)
      puts @@juego.jugadorActual.to_s << "\n"
      
      @@juego.siguienteJugador
      puts @@juego.jugadorActual.to_s << "\n"
      @@juego.mover(2)
      puts @@juego.jugadorActual.to_s << "\n"
      
      @@juego.siguienteJugador
      
      for i in 0..10 do
        if(@@juego.jugadorActual.casillaActual.numeroCasilla == 0)
          @@juego.mover(10)
        end
        if(@@juego.jugadorActual.casillaActual.numeroCasilla != 0)
          @@juego.mover(0)
        end
      end
      
      puts @@juego.jugadorActual.to_s << "\n"
      @@juego.mover(12)
      puts @@juego.jugadorActual.to_s << "\n"
      
      @@juego.siguienteJugador
      
      puts @@juego.jugadorActual.to_s << "\n"
      @@juego.mover(12)
      puts @@juego.jugadorActual.to_s << "\n"
      
      #Imprimimos por pantalla el tablero (casillas y cárcel)
      puts "\nEl juego es: " << @@juego.to_s
      
    end
  end
  PruebaQytetet.main
end

File: Universidad de Granada/Programación y Diseño Orientada a Objetos/QytetetJava/src/modeloqytetet/PruebaQytetet.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package modeloqytetet;
import java.util.ArrayList;
import java.util.Scanner;

/**
 *
 * @author Miguel Ángel Benítez Alguacil
 *         Najib Saadouli Arco
 */
public class PruebaQytetet {
    private static Qytetet juego = Qytetet.getQytetet();
    
    //Método1 muestra las sorpresas cuyo valor sea mayor que 0
    private static ArrayList<Sorpresa> Metodo1(){       
        ArrayList<Sorpresa> metodo = new ArrayList<>();     //Creamos un array de sorpresas que contendrá las cartas sorpresa que buscamos
        
        for (Sorpresa carta: juego.getMazo())               //Recorre todo el array cogiendo el valor de cada sorpresa del mazo en cada iteración
            if (carta.getValor() > 0)                       //Compara el valor de la sorpresa para ver si es mayor que 0
                metodo.add(carta);
        
        return metodo;
    }
    
    //Metodo2 muestra las sorpresas de tipo IRACASILLA
    private static ArrayList<Sorpresa> Metodo2(){
        ArrayList<Sorpresa> metodo = new ArrayList<>();
        
        for (Sorpresa carta: juego.getMazo())           
            if (carta.getTipo() == TipoSorpresa.IRACASILLA)     //Compara si el tipo de la sorpresa es IRACASILLA
                metodo.add(carta);
        
        return metodo;
    }
    
    //Metodo3 muestra las sorpresas del tipo que se le indica en el argumento
    private static ArrayList<Sorpresa> Metodo3(TipoSorpresa tipo){
        ArrayList<Sorpresa> metodo = new ArrayList<>();
        
        for (Sorpresa carta: juego.getMazo())           
            if (carta.getTipo() == tipo)                    //Compara los tipos de las sorpresas y si es igual al del argumento la añade al array
                metodo.add(carta);
        
        return metodo;
    }
    
    //Método para introducir el nombre de los jugadores desde teclado
    private static ArrayList<String> getNombreJugadores(){
        Scanner capture = new Scanner(System.in);
        int numeroJugadores = 0;
        
        do{
            System.out.print("¿Cuál es el número de jugadores?: ");
            numeroJugadores = capture.nextInt();
        }while(numeroJugadores < 2 || numeroJugadores > Qytetet.MAX_JUGADORES);
        
        ArrayList<String> nombres = new ArrayList<>();
        
        for(int i=0; i<numeroJugadores; i++){
            System.out.print("\tJugador "+ i + ": ");
            nombres.add(capture.next());
        }        
        
        return nombres;
    }
   
    //***MAIN***
    public static void main(String[] args) {
        ArrayList<String> nombres = PruebaQytetet.getNombreJugadores();
        
        juego.inicializarJuego(nombres);
        
        /*
        System.out.print("\nMétodo1 -> Las sorpresas con valores mayores que 0 son: ");
        for (Sorpresa carta: Metodo1())             //Bucle que recorre el mazo de sorpresas y realiza el método 1
            System.out.print("\t" + carta.toString());
        
        System.out.print("\n\nMétodo2 -> Las sorpresas para ir a casilla son: ");
        for (Sorpresa carta: Metodo2())             //Bucle que recorre el mazo de sorpresas y realiza el método 2
            System.out.print("\t" + carta.toString());
        
        System.out.print("\n\nMétodo3 -> Las sorpresas de cada tipo son: ");
        for (TipoSorpresa carta: TipoSorpresa.values()){      //Con TipoSorpresa.values() obtenemos un array con todos los valores posibles de TipoSorpresa y carta va obteniendo cada uno de los valores
            System.out.print("Tipo de Sorpresa: " + carta.toString() );
            for(Sorpresa y: Metodo3(carta))          //Bucle que recorre el mazo de sorpresas y realiza el método 3 para cada uno de los valores de carta
                System.out.print("\t" + y.toString());
        }
        
        
        Scanner in = new Scanner (System.in);
        int test = 0 ;
        
        System.out.print("\nOPCCIONES PARA EL TEST DE LA PRACTICA 3: "
                + "\n 1) Prueba metodo mover y comprar \n "
                + "\n 2) Prueba metodo comprar y pagarAlquiler \n "
                + "\n 3) Prueba metodo aplicarSorpresa \n "
                +  "\n 4) Prueba hipotecar, cancelar,vender y edificar casas y hoteles\n"
                +  "\n 5) Prueba a salir de carcel\n"
                +  "\n 6) Prueba de ranking y simulacion de juego");
        
        System.out.print("\nIntroduzca la opccion para su prueba: \n");
        
        test = in.nextInt(); in.skip("\n");
        
        if(test == 1)
        {
            System.out.print("\n\nSalida del jugador \n"); 
            System.out.println(juego.getJugadorActual().getCasillaActual().getTipo());
            System.out.print("\nMoviendo jugador a casilla 5 de calle y comprandola \n"); 
            juego.mover(5);
            System.out.println(juego.getJugadorActual().getCasillaActual().getTipo());
            System.out.println(juego.getEstado()); 
             if(juego.getEstado() == EstadoJuego.JA_PUEDECOMPRAROGESTIONAR)
                 juego.comprarTituloPropiedad();  
            System.out.println(juego.getEstado());  
            System.out.print("\nJugador ahora en calle impuesto \n");            
            juego.mover(2);
            System.out.println(juego.getJugadorActual().getCasillaActual().getTipo());
            System.out.println(juego.getEstado());             
            System.out.print("\nJugador ahora en calle sorpresa \n");            
            juego.mover(3);
            System.out.println(juego.getJugadorActual().getCasillaActual().getTipo());
            System.out.println(juego.getEstado()); 
            System.out.print("\nMostrando propiedades de jugador \n");            
            System.out.print(juego.getJugadorActual().getPropiedades());
        }
        
        if(test == 2)
        {
            System.out.print("\nMoviendo jugador a casilla 5 de calle y comprandola");             
            System.out.print("\n"+ juego.getJugadorActual().getNombre() + "\n");                        
            juego.mover(5);
            System.out.println(juego.getJugadorActual().getCasillaActual().getTipo());
            System.out.println(juego.getEstado()); 
            System.out.print("Saldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");                        

             if(juego.getEstado() == EstadoJuego.JA_PUEDECOMPRAROGESTIONAR)
                 System.out.println(juego.comprarTituloPropiedad()); 

             System.out.print("\nMoviendo jugador a casilla 5 de calle y comprandola \n");             

             juego.siguienteJugador();
             System.out.print("\n"+ juego.getJugadorActual().getNombre() + "\n");                        
             System.out.print("\nSaldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");                        
             juego.mover(8);
             System.out.println(juego.getEstado()); 
             System.out.print("\nSaldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");                        
             juego.siguienteJugador();
             System.out.print("\n"+ juego.getJugadorActual().getNombre() + "\n");                        
             System.out.print("\nSaldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");                               
        }
        
        if(test == 3)
        {
            System.out.println("\nVamos a recorrer todo el mazo, aplicando así todas las sorpresas: ");
                for(int i = 0; i < Qytetet.NUM_SORPRESAS; i++)
                {
                    juego.mover(3);
                    System.out.print("\nSORPRESA Nº: " + i + " tipo: " + juego.getCartaActual().getTipo());
                    juego.aplicarSorpresa();
                    System.out.print("\nEstado del juego: ");
                    System.out.println(juego.getEstado());  
                    juego.siguienteJugador();
                } 
        }
        
        if(test == 4)
        {         
            
            System.out.print("\nMoviendo jugador a casilla 4 de calle y comprandola\n");             
            juego.mover(4);
            System.out.println(juego.getJugadorActual().getCasillaActual().getTipo());
            System.out.println(juego.getEstado()); 
            System.out.print("Saldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");
            System.out.print("\nEdificando casas y edificando hotel\n");  
            juego.comprarTituloPropiedad();
            juego.edificarCasa(4);
            juego.edificarCasa(4);
            juego.edificarCasa(4);

            System.out.println("numero de casas = " + juego.getJugadorActual().getPropiedades().get(0).getNumCasas()) ;

            juego.edificarCasa(4);
            juego.edificarHotel(4);

            System.out.print("\nEdificado un hotel \n");  
            System.out.println("numero de casas = " + juego.getJugadorActual().getPropiedades().get(0).getNumCasas()) ;
            System.out.println("numero de hoteles = " + juego.getJugadorActual().getPropiedades().get(0).getNumHoteles()) ;

            System.out.print("Saldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");
            System.out.println("\nHipotecar\n");
            juego.hipotecarPropiedad(4);
            System.out.println("hipotecado :" + juego.getJugadorActual().getCasillaActual().getTitulo().getHipotecada());
            System.out.print("Saldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");  

            System.out.println("\nCancelando hipoteca");
            juego.cancelarHipoteca(4);
            System.out.println("hipotecado :" + juego.getJugadorActual().getCasillaActual().getTitulo().getHipotecada());
            System.out.print("Saldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");

            System.out.println("\nVenta propiedad");
            juego.venderPropiedad(4);
            System.out.println("propietario" + juego.getJugadorActual().getCasillaActual().getTitulo().tengoPropietario());
            System.out.print("Saldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");
           
        }
        
        if(test == 5)
        {
            System.out.print("\nSe mete al jugador en la carcel\n");
            juego.mover(12);
            System.out.println("encarcelado = " + juego.getJugadorActual().getEncarcelado());
            juego.intentarSalirCarcel(MetodoSalirCarcel.PAGANDOLIBERTAD);
            System.out.println("encarcelado = " + juego.getJugadorActual().getEncarcelado());
            System.out.print("Saldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");

            boolean fuera = false;
            System.out.print("\nSe vuelve a meter al jugador en la carcel\n");
            juego.mover(12);
            System.out.println("encarcelado = " + juego.getJugadorActual().getEncarcelado());
            while(!fuera)
            {
                juego.intentarSalirCarcel(MetodoSalirCarcel.TIRANDODADO);
                if(!juego.getJugadorActual().getEncarcelado())
                    fuera = true ;
            }           
            System.out.println("encarcelado = " + juego.getJugadorActual().getEncarcelado());
            System.out.print("Saldo jugador = "+ juego.getJugadorActual().getSaldo() + "\n");
        }
        
        if(test == 6)
        {
            for(int i = 0; i < 30; i++)
                {
                    juego.jugar();
                    juego.siguienteJugador();
                }
                juego.mover(2);
                juego.siguienteJugador();
                juego.mover(1);
                juego.comprarTituloPropiedad();
                juego.hipotecarPropiedad(1);
                juego.siguienteJugador();
                juego.mover(6);
                juego.comprarTituloPropiedad();
                juego.siguienteJugador();
                juego.mover(4);
                juego.comprarTituloPropiedad();
                juego.siguienteJugador();
                juego.mover(3);
                juego.aplicarSorpresa();
                juego.mover(2);
                juego.siguienteJugador();
                juego.mover(2);
                juego.siguienteJugador();
                juego.obtenerRanking();
                for(Jugador j: juego.getJugadores())
                    System.out.println(j);
        }
        */
        
        juego.mover(3);
        
        System.out.print(juego.getJugadorActual() + "\n");
        
        juego.aplicarSorpresa();
        
        System.out.print(juego.getJugadorActual() + "\n");
        
        for(int i=10; i>0; i--){
            if(juego.getJugadorActual().getCasillaActual().getNumeroCasilla() == 0)
                juego.mover(10);
            if(juego.getJugadorActual().getCasillaActual().getNumeroCasilla() != 0)
                juego.mover(0);
        }
        
        juego.mover(4);
        
        if(juego.comprarTituloPropiedad())
            System.out.print(juego.getJugadorActual() + "\n");
        
        for(int i=0; i<10; i++)
            if(juego.edificarCasa(4))
                System.out.print(juego.getJugadorActual() + "\n");
        
        for(int i=0; i<10; i++){
            if(juego.edificarHotel(4))
                System.out.print(juego.getJugadorActual() + "\n");
            for(int j=0; j<10; j++)
                if(juego.edificarCasa(4))
                    System.out.print(juego.getJugadorActual() + "\n");
        }
        
        juego.siguienteJugador();
        
        System.out.print(juego.getJugadorActual() + "\n");
        
        for(int i=10; i>0; i--){
            if(juego.getJugadorActual().getCasillaActual().getNumeroCasilla() == 0)
                juego.mover(10);
            if(juego.getJugadorActual().getCasillaActual().getNumeroCasilla() != 0)
                juego.mover(0);
        }
        
        juego.mover(5);
        
        if(juego.comprarTituloPropiedad())
            System.out.print(juego.getJugadorActual() + "\n");
        
        for(int i=0; i<10; i++)
            if(juego.edificarCasa(5))
                System.out.print(juego.getJugadorActual() + "\n");
        
        for(int i=0; i<10; i++){
            if(juego.edificarHotel(5))
                System.out.print(juego.getJugadorActual() + "\n");
            for(int j=0; j<10; j++)
                if(juego.edificarCasa(5))
                    System.out.print(juego.getJugadorActual() + "\n");
        }
        
        juego.siguienteJugador();
        System.out.print(juego.getJugadorActual() + "\n");
        juego.mover(2);
        System.out.print(juego.getJugadorActual() + "\n");
        
        juego.siguienteJugador();
        System.out.print(juego.getJugadorActual() + "\n");
        juego.mover(2);
        System.out.print(juego.getJugadorActual() + "\n");
        
        juego.siguienteJugador();
        
        for(int i=10; i>0; i--){
            if(juego.getJugadorActual().getCasillaActual().getNumeroCasilla() == 0)
                juego.mover(10);
            if(juego.getJugadorActual().getCasillaActual().getNumeroCasilla() != 0)
                juego.mover(0);
        }
        
        System.out.print(juego.getJugadorActual() + "\n");
        juego.mover(12);
        System.out.print(juego.getJugadorActual() + "\n");
        
        juego.siguienteJugador();
        
        System.out.print(juego.getJugadorActual() + "\n");
        juego.mover(12);
        System.out.print(juego.getJugadorActual() + "\n");
        
        System.out.println(juego);
    }
}



